### ISSUE TITLE
Docker internal load balancer intermittent connectivity issue

### ISSUE THREAD

**Issue by getvivekv** [moby/moby#35671]
Having this issues when trying to connect from one container to another container using the swarm load balancer's internal name.

Both the containers are on the same host, same overlay network.

Container 1 name: us-int-sl
Container 2 name: us-int-soss

Trying to connect from 1 to 2 works only half of the time

```
root@us-int-sl:~# curl -v us-int-soss
* Rebuilt URL to: us-int-soss/
*   Trying 10.0.0.6...
* Connected to us-int-soss (10.0.0.6) port 80 (#0)
> GET / HTTP/1.1
> Host: us-int-soss
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 302 Found
< Date: Fri, 01 Dec 2017 18:18:54 GMT
< Server: Apache/2.4.6 (CentOS) PHP/5.3.28 OpenSSL/1.0.1e-fips
< X-Powered-By: PHP/5.3.28
< Set-Cookie: EASSESSID=u6rmu6ktmv8clf57m5ct3hni74; path=/
< Expires: Mon, 26 Jul 1997 05:00:00 GMT
< Cache-Control: no-store, no-cache, must-revalidate
< Pragma: no-cache
< Last-Modified: Fri, 01 Dec 2017 18:18:54 GMT
< Cache-Control: post-check=0, pre-check=0
< Set-Cookie: eas_uuid=f801ae966499a4071b8134ae3e6b48af; expires=Sat, 01-Dec-2018 18:18:54 GMT; path=/
< Set-Cookie: eas_uuid=f801ae966499a4071b8134ae3e6b48af; expires=Sat, 01-Dec-2018 18:18:54 GMT; path=/
< Location: /clear_session.esp
< Content-Length: 0
< Content-Type: text/html; charset=UTF-8
< 
* Connection #0 to host us-int-soss left intact


root@us-int-sl:~# 


root@us-int-sl:~# curl -v us-int-soss
* Rebuilt URL to: us-int-soss/
*   Trying 10.0.0.6...
^C



root@us-int-sl:~# curl -v us-int-soss
* Rebuilt URL to: us-int-soss/
*   Trying 10.0.0.6...
* Connected to us-int-soss (10.0.0.6) port 80 (#0)
> GET / HTTP/1.1
> Host: us-int-soss
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 302 Found
< Date: Fri, 01 Dec 2017 18:19:04 GMT
< Server: Apache/2.4.6 (CentOS) PHP/5.3.28 OpenSSL/1.0.1e-fips
< X-Powered-By: PHP/5.3.28
< Set-Cookie: EASSESSID=885ecp2kb98i7r4r2gl6pqqgg2; path=/
< Expires: Mon, 26 Jul 1997 05:00:00 GMT
< Cache-Control: no-store, no-cache, must-revalidate
< Pragma: no-cache
< Last-Modified: Fri, 01 Dec 2017 18:19:04 GMT
< Cache-Control: post-check=0, pre-check=0
< Set-Cookie: eas_uuid=2303e1545de31c52d8a2da7104d31c28; expires=Sat, 01-Dec-2018 18:19:04 GMT; path=/
< Set-Cookie: eas_uuid=2303e1545de31c52d8a2da7104d31c28; expires=Sat, 01-Dec-2018 18:19:04 GMT; path=/
< Location: /clear_session.esp
< Content-Length: 0
< Content-Type: text/html; charset=UTF-8
< 
* Connection #0 to host us-int-soss left intact



root@us-int-sl:~# curl -v us-int-soss
* Rebuilt URL to: us-int-soss/
*   Trying 10.0.0.6...
* connect to 10.0.0.6 port 80 failed: Connection timed out
* Failed to connect to us-int-soss port 80: Connection timed out
* Closing connection 0
curl: (7) Failed to connect to us-int-soss port 80: Connection timed out




root@us-int-sl:~# curl -v us-int-soss
* Rebuilt URL to: us-int-soss/
*   Trying 10.0.0.6...
* Connected to us-int-soss (10.0.0.6) port 80 (#0)
> GET / HTTP/1.1
> Host: us-int-soss
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 302 Found
< Date: Fri, 01 Dec 2017 18:21:59 GMT
< Server: Apache/2.4.6 (CentOS) PHP/5.3.28 OpenSSL/1.0.1e-fips
< X-Powered-By: PHP/5.3.28
< Set-Cookie: EASSESSID=6cc7r2bui3gjco2bkmr3prelf1; path=/
< Expires: Mon, 26 Jul 1997 05:00:00 GMT
< Cache-Control: no-store, no-cache, must-revalidate
< Pragma: no-cache
< Last-Modified: Fri, 01 Dec 2017 18:21:59 GMT
< Cache-Control: post-check=0, pre-check=0
< Set-Cookie: eas_uuid=0481d58cca5aea1d4e3893794d1a8ca4; expires=Sat, 01-Dec-2018 18:21:59 GMT; path=/
< Set-Cookie: eas_uuid=0481d58cca5aea1d4e3893794d1a8ca4; expires=Sat, 01-Dec-2018 18:21:59 GMT; path=/
< Location: /clear_session.esp
< Content-Length: 0
< Content-Type: text/html; charset=UTF-8
< 
* Connection #0 to host us-int-soss left intact
root@us-int-sl:~# 

```

```
root@us-n-w-1:~# docker --version
Docker version 17.06.0-ce, build 02c1d87
root@us-n-w-1:~# 

```

```
root@us-n-w-1:~# docker info
Containers: 18
 Running: 0
 Paused: 0
 Stopped: 18
Images: 35
Server Version: 17.06.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 375
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: pending
 NodeID: vepwmpywjpmq9cvi8o7g7p2d1
 Is Manager: false
 Node Address: 172.24.19.119
 Manager Addresses:
  172.24.19.127:2377
  172.24.19.192:2377
  172.24.19.230:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-1022-aws
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.67GiB
Name: us-n-w-1
ID: 4WIP:QCE6:27HQ:VJT4:7UIV:42EG:SU3I:BX2T:DOIO:IUBX:3HXO:RAF4
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
root@us-n-w-1:~# 

```

**Comment by thaJeztah** [comment]
It looks like you're running an outdated version of docker; Docker 17.06.0-CE reached end of life in October; if you cannot update to a more recent version of Docker, make sure you're running the latest patch release for the 17.06 release (which is 17.06.2-CE).

If you're still able to reproduce this issue on a current release of Docker, could you provide

- the exact steps to reproduce; e.g. in your example you talk about "containers"; are these containers started manually, or part of a service (`docker service create`)?
- what does your cluster look like? Are all nodes in a single data-centre, multiple data-centers? Is docker running on bare-metal? A specific cloud provider you're using?
- are you using `-o encrypted` for your overlay networks?
- is there anything in the logs that stands out?

**Comment by mateuszpodraza** [comment]
@getvivekv Is this a fresh setup or have you scaled your services up and down, removed/added some nodes? 

You're having the same symptoms that we did and they eventually led us to discover that #35548 is the culprit. You can try running ``docker network inspect -v network_name`` and see if the tasks reported to be on that network still exist.

**Comment by DylanSp** [comment]
I'm a coworker of @getvivekv, working on the same issue.
@thaJeztah We'll try upgrading Docker to 17.09; we just need to test it first to make sure it doesn't break things further.
@mateuszpodraza From running `docker network inspect -v us-int` (`us-int-sl` and `us-int-soss` are both attached to that network), I see dead tasks; that may well be causing the issue. Thanks for the info.

**Comment by getvivekv** [comment]
@thaJeztah These containers are started with docker service create. The cluster consists of 3 managers and 5 workers in AWS autoscale (custom designed). Deployments happen quite a few times in this cluster hence a lot of 'docker service update' happens daily. 

Any clue of the dead tasks and how to get rid of them other than destroying the entire cluster and rebuild?

**Comment by getvivekv** [comment]
@thaJeztah @mavenugo  Upgraded to `Docker version 17.06.2-ce, build cec0b72`

We are still receiving this error. A container is unable to connect to other containers intermittently. Is there any additional troubleshooting we could do to check what's the issue? 

Just also tried to replace overlay network with Weave net plugin thinking that it could be an overlay issue (https://www.weave.works/oss/net/) but I could replicate the same issue there as well. Looks like an issue with docker+iptables

**Comment by getvivekv** [comment]
@thaJeztah  Here are the answers to the questions 


the exact steps to reproduce; e.g. in your example you talk about "containers"; are these containers started manually, or part of a service (docker service create)?

- No specific steps to reproduce. It happens intermittently. Most likely after few docker update commands
- Containers started using docker service create 


what does your cluster look like? Are all nodes in a single data-centre, multiple data-centers? Is docker running on bare-metal? A specific cloud provider you're using?

- 3 manages, 4 worker nodes
- On AWS 
- Tested with all servers in SAME datacenter and also multiple datacenters. Problem appears to be on both.


are you using -o encrypted for your overlay networks?

- No. Even attempted to use Weavenet instead of overlay. 

is there anything in the logs that stands out?

- Nothing that is suspicious.

**Comment by thaJeztah** [comment]
Thanks! Nothing that stands out from that 😥

**Comment by getvivekv** [comment]
Is there any additional info needed from me on this ?

**Comment by Mobe91** [comment]
@thaJeztah 
I just stumbled over this and I have also found another [misplaced issue](https://github.com/docker/swarm/issues/2855) that looks very alike.

I am running on a single node swarm cluster running 13 services. The symptom started when the whole set of services suddenly restarted all at once - I suspect the restart to have been caused by an overload of the server due to another non-swarm docker container that was running an expensive build on the same host at the time. All service tasks were able to get back into RUNNING state fine but then the connectivity issues kicked in.

One examplary outtake:
There is a frontend reverse proxy service (nginx) and a backend website service (nginx). Both services are attached to the `nginx` overlay network. When I `docker exec` into the reverse proxy and `curl http://website-service`, _precisely_ every second request fails.

Output of `docker network inspect -v nginx`:
```
[
    {
        "Name": "nginx",
        "Id": "yqxbrgtnrvvj3xgq2mitgkq2i",
        "Created": "2018-04-24T14:03:59.553151529+02:00",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "10.0.0.0/24",
                    "Gateway": "10.0.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "1aef4fbb9746dd983f47fe1bed0987cafb97d414c4dd772d630886d0639bffd8": {
                "Name": "ordami-backend-staging_ordami-backend.1.krztg2xoyr0w7xmpxdtwklcpy",
                "EndpointID": "ba44f5f7b9a2b8774a5c4b1cc600b692aba0b25c670fc7cf74110a579caa52f2",
                "MacAddress": "02:42:0a:00:00:08",
                "IPv4Address": "10.0.0.8/24",
                "IPv6Address": ""
            },
            "259b81b1d6e7b3ea3ff585981980487c94e48351b5454b3fad2a84c6f82daa59": {
                "Name": "keycloak-staging_keycloak.1.dgic1hqsuv783qsk8r2grwrec",
                "EndpointID": "ccc5d85ac00531a76792097ec4b1405a27c66a4218c4479bacf78afdcd07bab9",
                "MacAddress": "02:42:0a:00:00:13",
                "IPv4Address": "10.0.0.19/24",
                "IPv6Address": ""
            },
            "37e76b7345044984c78f57a670a04bff0c84b7e814d2c1a9c6fa64c12d09a0e6": {
                "Name": "graphhopper-staging_graphhopper.1.nklovdlg84gjveoc4ngwjopfn",
                "EndpointID": "31d959d474f078b82c10cdf69ec0ceda4b8775eda737095c1d6727bafc58e251",
                "MacAddress": "02:42:0a:00:00:05",
                "IPv4Address": "10.0.0.5/24",
                "IPv6Address": ""
            },
            "407fc9adf0b72a0b06837dc213248caa38d22e3c8e99babda8c9521c4980b40d": {
                "Name": "ordami-website-staging_ordami-website.1.amiyiduocyafjybuklwraz1y5",
                "EndpointID": "f9016203657a8fb3b4f79d67412d2dba598d3384418f5f82b80588dc0191d956",
                "MacAddress": "02:42:0a:00:00:fc",
                "IPv4Address": "10.0.0.252/24",
                "IPv6Address": ""
            },
            "61b7a36542596d59824358d5cdfe7dd79dc1560cf8c51882eb7fdeb2d8b9b389": {
                "Name": "ordami-geo-staging_ordami-geo.1.zgnmvb8vssa81yayibhkkq73s",
                "EndpointID": "d112643daa1fbc6eb0b44fa39c779813c4e235e6b54df4c500caa497a29f170d",
                "MacAddress": "02:42:0a:00:00:15",
                "IPv4Address": "10.0.0.21/24",
                "IPv6Address": ""
            },
            "7923c7b5f4c41602d7d11646c5538cedbb8b71e4dcf20dc84bfa13ed91d75a8d": {
                "Name": "nominatim-staging_nominatim.1.mqzdwh98mcoad23s2u9vj0zw6",
                "EndpointID": "44ab9c4c3e5f7d536271abf40e760dad368f66c67a319a57f6a8c048a3283057",
                "MacAddress": "02:42:0a:00:00:fb",
                "IPv4Address": "10.0.0.251/24",
                "IPv6Address": ""
            },
            "b39f785323f6c107b96fa3702e7c4cc54d1be19bf7634b554a9dd35a60924f1e": {
                "Name": "nginx_nginx.1.qnubwz5swl5554z1wx4duwbka",
                "EndpointID": "a7d67a3c0bf3b9bb3f104cb5c134417ae47e81f9023df68e64d90be528ec6621",
                "MacAddress": "02:42:0a:00:00:18",
                "IPv4Address": "10.0.0.24/24",
                "IPv6Address": ""
            },
            "c15c47ba67d3f5806acee67e3f70aa47e0649be57a0d866de997fc0b175b73b0": {
                "Name": "ordami-backoffice-staging_ordami-backoffice.1.x18z5qwukodfjngod8qaj2b9p",
                "EndpointID": "5757e2054c6ea4e3892b1101336642e51ff4fce257478dc360bf652dda8e48ea",
                "MacAddress": "02:42:0a:00:00:06",
                "IPv4Address": "10.0.0.6/24",
                "IPv6Address": ""
            },
            "f3ac77223c336dc52951b7ab96de8178aa431064f794166f45fcc2d867b17d5a": {
                "Name": "ordami-wko-staging_ordami-wko-api.1.5u4wr3dlxiaycridij5t14fk4",
                "EndpointID": "92ba2fe81086b43fb2c785b0f09def47491348da272c72b6231e1f14ed6b8df6",
                "MacAddress": "02:42:0a:00:00:12",
                "IPv4Address": "10.0.0.18/24",
                "IPv6Address": ""
            },
            "f80b68cea5796d323d06a68ed6a82bc13e2a4780be758b01b2710df4bf7d5e2a": {
                "Name": "ordami-wko-staging_ordami-wko-sync.1.v4xuba5i5uzfndxlyqxwita3i",
                "EndpointID": "7133f90c94fcc5141b26e2388651f5136b41f593703ee9eb4e187e81f263f149",
                "MacAddress": "02:42:0a:00:00:10",
                "IPv4Address": "10.0.0.16/24",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "4097"
        },
        "Labels": {},
        "Peers": [
            {
                "Name": "4912820cfa0d",
                "IP": "185.242.113.117"
            }
        ],
        "Services": {
            "graphhopper-staging_graphhopper": {
                "VIP": "10.0.0.162",
                "Ports": [],
                "LocalLBIndex": 1719,
                "Tasks": [
                    {
                        "Name": "graphhopper-staging_graphhopper.1.nklovdlg84gjveoc4ngwjopfn",
                        "EndpointID": "31d959d474f078b82c10cdf69ec0ceda4b8775eda737095c1d6727bafc58e251",
                        "EndpointIP": "10.0.0.5",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    }
                ]
            },
            "keycloak-staging_keycloak": {
                "VIP": "10.0.0.7",
                "Ports": [],
                "LocalLBIndex": 1724,
                "Tasks": [
                    {
                        "Name": "keycloak-staging_keycloak.1.dgic1hqsuv783qsk8r2grwrec",
                        "EndpointID": "ccc5d85ac00531a76792097ec4b1405a27c66a4218c4479bacf78afdcd07bab9",
                        "EndpointIP": "10.0.0.19",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    }
                ]
            },
            "nginx_nginx": {
                "VIP": "10.0.0.172",
                "Ports": [],
                "LocalLBIndex": 1676,
                "Tasks": [
                    {
                        "Name": "nginx_nginx.1.y2udchmurgfoid2y88wmg9g8r",
                        "EndpointID": "1c337539e7b64d51ffc5f2c31dfed8d1cb3d0ef85a40d4575d680f1b38f0890c",
                        "EndpointIP": "10.0.0.153",
                        "Info": null
                    },
                    {
                        "Name": "nginx_nginx.1.qnubwz5swl5554z1wx4duwbka",
                        "EndpointID": "a7d67a3c0bf3b9bb3f104cb5c134417ae47e81f9023df68e64d90be528ec6621",
                        "EndpointIP": "10.0.0.24",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    },
                    {
                        "Name": "nginx_nginx.1.ttem68yh9b0hcypkl0jzuwvge",
                        "EndpointID": "a5a61e22a41e25c9f161f95490d12c25fc1269bd68b16901100e31c844347374",
                        "EndpointIP": "10.0.0.35",
                        "Info": null
                    },
                    {
                        "Name": "nginx_nginx.1.s8u54ay0k7qm1jnigr21j5bpf",
                        "EndpointID": "612191f32a70894ee025705dd9de6e157ab60fedb8f4cd6201f2ea69e3972f87",
                        "EndpointIP": "10.0.0.102",
                        "Info": null
                    }
                ]
            },
            "nominatim-staging_nominatim": {
                "VIP": "10.0.0.11",
                "Ports": [],
                "LocalLBIndex": 1701,
                "Tasks": [
                    {
                        "Name": "nominatim-staging_nominatim.1.mqzdwh98mcoad23s2u9vj0zw6",
                        "EndpointID": "44ab9c4c3e5f7d536271abf40e760dad368f66c67a319a57f6a8c048a3283057",
                        "EndpointIP": "10.0.0.251",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    }
                ]
            },
            "ordami-backend-staging_ordami-backend": {
                "VIP": "10.0.0.13",
                "Ports": [],
                "LocalLBIndex": 1705,
                "Tasks": [
                    {
                        "Name": "ordami-backend-staging_ordami-backend.1.krztg2xoyr0w7xmpxdtwklcpy",
                        "EndpointID": "ba44f5f7b9a2b8774a5c4b1cc600b692aba0b25c670fc7cf74110a579caa52f2",
                        "EndpointIP": "10.0.0.8",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    }
                ]
            },
            "ordami-backoffice-staging_ordami-backoffice": {
                "VIP": "10.0.0.15",
                "Ports": [],
                "LocalLBIndex": 1702,
                "Tasks": [
                    {
                        "Name": "ordami-backoffice-staging_ordami-backoffice.1.x18z5qwukodfjngod8qaj2b9p",
                        "EndpointID": "5757e2054c6ea4e3892b1101336642e51ff4fce257478dc360bf652dda8e48ea",
                        "EndpointIP": "10.0.0.6",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    }
                ]
            },
            "ordami-geo-staging_ordami-geo": {
                "VIP": "10.0.0.237",
                "Ports": [],
                "LocalLBIndex": 1722,
                "Tasks": [
                    {
                        "Name": "ordami-geo-staging_ordami-geo.1.zgnmvb8vssa81yayibhkkq73s",
                        "EndpointID": "d112643daa1fbc6eb0b44fa39c779813c4e235e6b54df4c500caa497a29f170d",
                        "EndpointIP": "10.0.0.21",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    }
                ]
            },
            "ordami-website-staging_ordami-website": {
                "VIP": "10.0.0.17",
                "Ports": [],
                "LocalLBIndex": 1677,
                "Tasks": [
                    {
                        "Name": "ordami-website-staging_ordami-website.1.vjg0m418aegmhvxrolzlalb6b",
                        "EndpointID": "aaabf050c9b2d91e9aa9635019f2b841b40a6890e743877e2af0e113b060967f",
                        "EndpointIP": "10.0.0.204",
                        "Info": null
                    },
                    {
                        "Name": "ordami-website-staging_ordami-website.1.amiyiduocyafjybuklwraz1y5",
                        "EndpointID": "f9016203657a8fb3b4f79d67412d2dba598d3384418f5f82b80588dc0191d956",
                        "EndpointIP": "10.0.0.252",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    }
                ]
            },
            "ordami-wko-staging_ordami-wko-api": {
                "VIP": "10.0.0.23",
                "Ports": [],
                "LocalLBIndex": 1713,
                "Tasks": [
                    {
                        "Name": "ordami-wko-staging_ordami-wko-api.1.5u4wr3dlxiaycridij5t14fk4",
                        "EndpointID": "92ba2fe81086b43fb2c785b0f09def47491348da272c72b6231e1f14ed6b8df6",
                        "EndpointIP": "10.0.0.18",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    }
                ]
            },
            "ordami-wko-staging_ordami-wko-sync": {
                "VIP": "10.0.0.26",
                "Ports": [],
                "LocalLBIndex": 1714,
                "Tasks": [
                    {
                        "Name": "ordami-wko-staging_ordami-wko-sync.1.v4xuba5i5uzfndxlyqxwita3i",
                        "EndpointID": "7133f90c94fcc5141b26e2388651f5136b41f593703ee9eb4e187e81f263f149",
                        "EndpointIP": "10.0.0.16",
                        "Info": {
                            "Host IP": "185.242.113.117"
                        }
                    }
                ]
            }
        }
    }
]
```
Observe that the network is reported to have the following dead tasks attached:
* ordami-website-staging_ordami-website.1.vjg0m418aegmhvxrolzlalb6b
* nginx_nginx.1.y2udchmurgfoid2y88wmg9g8r
* nginx_nginx.1.ttem68yh9b0hcypkl0jzuwvge
* nginx_nginx.1.s8u54ay0k7qm1jnigr21j5bpf

`ordami-website-staging_ordami-website` is the website service that I was talking about before. If we consider that the `nginx` network as shown above has 2 task entries for this service, one of which is dead, it becomes plausible that every second request to the service fails as swarm tries to load balance the requests to all service tasks in a round robin fashion.

I have also attached the full [/var/log/syslog](https://github.com/moby/moby/files/2032008/syslog.zip) as it exhibits some docker network related logging. It should also cover the full service restart timespan (this must have happened at around 17:10 if I am not mistaken).

As reference for the network IDs contained in the log, here is the output of `docker network ls`.
```
NETWORK ID          NAME                                DRIVER              SCOPE
b6b3ba831074        bridge                              bridge              local
8c053423d479        docker_gwbridge                     bridge              local
rr5ymol86u7w        graphhopper-staging_graphhopper     overlay             swarm
972165952fb9        host                                host                local
6ijpz2obzmik        ingress                             overlay             swarm
7wzac8hyvlzn        keycloak-staging_db                 overlay             swarm
yqxbrgtnrvvj        nginx                               overlay             swarm
80185a17ccba        none                                null                local
j0nyaks2bgmu        ordami-backend-staging_db           overlay             swarm
4lu2dhh71dp6        ordami-wko-staging_db               overlay             swarm
eo55g2yytxgi        ordami-wko-staging_ordami-wko-api   overlay             swarm
```

# Environment Information
**docker version**
```
Client:
 Version:       18.03.0-ce
 API version:   1.37
 Go version:    go1.9.4
 Git commit:    0520e24
 Built: Wed Mar 21 23:10:01 2018
 OS/Arch:       linux/amd64
 Experimental:  false
 Orchestrator:  swarm

Server:
 Engine:
  Version:      18.03.0-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.4
  Git commit:   0520e24
  Built:        Wed Mar 21 23:08:31 2018
  OS/Arch:      linux/amd64
  Experimental: false
```
**Host OS**
```
Ubuntu 16.04.1 LTS
Linux ordami1 4.4.0-31-generic #50-Ubuntu SMP Wed Jul 13 00:07:12 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

I am happy to provide more information if needed.

**Comment by Mobe91** [comment]
# How I resolved this

I first removed the services which the network had dead tasks attached for (i.e. nginx_nginx and ordami-website-staging_ordami-website service).
After removing the 2 services, the nginx network still showed the dead tasks for nginx_nginx and ordami-website-staging_ordami-website, respectively. Only the "real"/alive tasks had been removed.

I thus tried to delete the nginx network altogether. This required me to remove all other attached services first. Interestingly, after removing the last attached service, also the dead tasks were gone in the output of `docker network inspect -v nginx`. I verified that the dead tasks were still listed _before_ removing the last service.

Since the existing network appeared to be "clean" now, I did not delete it but just recreated all the services. The recreated services worked fine and there were no more connectivity issues.

**Comment by luheorga** [comment]
I experienced the issue yesterday in a swarm cluster of 3 nodes, not all the services had that problem I tried to `curl` from various services to the bugged service, and many of them worked well, it's hard to follow the issue. 

**docker version**

```
Client: Docker Engine - Community
 Version:           20.10.4
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        d3cb89e
 Built:             Thu Feb 25 07:05:27 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.4
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Thu Feb 25 07:03:23 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by bsousaa** [comment]
It looks like this issue went stale. Let me close it.
